packages/project/src/db/style-sources.ts

```diff
@@ -0,0 +1,43 @@
+import { type Patch, applyPatches } from "immer";
+import { prisma } from "@webstudio-is/prisma-client";
+import {
+  authorizeProject,
+  type AppContext,
+} from "@webstudio-is/trpc-interface/server";
+import { StyleSources } from "@webstudio-is/project-build";
+import type { Build, Project } from "./schema";
+
+export const patch = async (
+  { buildId, projectId }: { buildId: Build["id"]; projectId: Project["id"] },
+  patches: Array<Patch>,
+  context: AppContext
+) => {
+  const canEdit = await authorizeProject.hasProjectPermit(
+    { projectId, permit: "edit" },
+    context
+  );
+
+  if (canEdit === false) {
+    throw new Error("You don't have edit access to this project");
+  }
+
+  const build = await prisma.build.findUnique({
+    where: { id: buildId },
+  });
+  if (build === null) {
+    return null;
```

Comment:
If you don't have to use null, I wouldn't use it, since its our level of abstraction, we could avoid it, idk mb this case a special so up to you
```suggestion
    return;
```